### ISSUE TITLE
Opening and saving PNG image results in Image being offset by X=1

### ISSUE THREAD

**Issue by devedse** [SixLabors/ImageSharp#288]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageSharp
- [X] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/JimBobSquarePants/ImageSharp/issuess) and [closed](https://github.com/JimBobSquarePants/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
When I load a certain image, all pixels seem to be moved 1 to the left:
![image](https://user-images.githubusercontent.com/2350015/28999561-ce3db9dc-7a4a-11e7-89ad-4503d33d856b.png)

### Steps to Reproduce
See the following code:
```
Directory.CreateDirectory(FolderHelperMethods.TempDirectoryForTests.Value);
var image1path = Path.Combine(FolderHelperMethods.AssemblyDirectoryForTests.Value, "TestImages", "vim16x16.png");
var outputImage = Path.Combine(FolderHelperMethods.TempDirectoryForTests.Value, "vim16x16output.png");

using (var img = Image.Load(image1path))
{
    using (var fs = new FileStream(outputImage, FileMode.Create))
    {
        img.SaveAsPng(fs);
    }

    //Pixel X=1 Y=0 should be transparent.
    var pixel = img.Pixels[1];
    Assert.True(pixel.A != 0); //This fails
}
```

Image that fails:
[vim16x16.zip](https://github.com/SixLabors/ImageSharp/files/1202766/vim16x16.zip)

I also created a failing unit test in and submitted a pull request: #289 

### System Configuration

- ImageSharp version: MyGet ImageSharp Version 1.0.0-alpha9-00175
- Other ImageSharp packages and versions: -
- Environment (Operating system, version and so on): Windows 10
- .NET Framework version: DotNet Core
- Additional information:

**Comment by JimBobSquarePants** [comment]
I wonder if #166 and the fix #167 are related to this?

**Comment by JimBobSquarePants** [comment]
Turns out it was a change to another method causing this. We were offsetting when we didn't need to anymore. 

Thanks for the PR with the test. That really helped and made the fix much easier to implement. 💯 

I'll get a build out asap with the fix, something has gone wrong with the transition to SixLabors.
